packages/react/src/components/Authenticator/FederatedIdentities/controls/createProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { CreateProviderInput } from './types';
+import { ElementsProvider } from '@aws-amplify/ui-react-core/elements';
+import {
+  ProviderDataListProvider,
+  HandleSignInWithRedirectProvider,
+  DisplayTextProvider,
+  FederatedIdentitiesElements,
+  UseHandleSignInWithRedirectContext,
+} from '../context';
+import { signInWithRedirect } from 'aws-amplify/auth';
+
+export default function createProvider<
+  T extends Partial<FederatedIdentitiesElements>,
+>({
+  elements,
+  providers,
+  handleSignInWithRedirect = signInWithRedirect,
+  displayText,
+}: CreateProviderInput<T>) {
+  return function Provider({
+    children,
+  }: {
+    children?: React.ReactNode;
+  }): React.JSX.Element {
+    const context = React.useContext(UseHandleSignInWithRedirectContext);
```

Comment:
Should this be `useHandleSignInWithRedirect`?

Reply:
I just didn't want an error to be thrown here from the useHandleSignInWithRedirect function. But I could do the function with a try/catch if you think that would be better.